### PR TITLE
Websocket name from session

### PR DESCRIPTION
**Pull request by jklingelAvid**
Added support for websocket names based on Session contents
Added missing charset param in PebbleBenchmark.scala

### CODE REVIEW

File: gatling-benchmarks/src/main/scala/io/gatling/core/session/pebble/PebbleBenchmark.scala [hunk 1]
```diff
@@ -31,10 +31,10 @@ object PebbleBenchmark {
   private val charset = configuration.core.charset
 
   private val session: Session = Session("Scenario", 0, null).set("id", 3)
-  private val SinglePlaceHolderPebbleBody = PebbleStringBody(resourceAsString("sample-peeble.json", charset))
+  private val SinglePlaceHolderPebbleBody = PebbleStringBody(resourceAsString("sample-peeble.json", charset), charset)
 
   private val session2: Session = Session("Scenario", 0, null).setAll("id" -> 3, "friends" -> Seq.fill(20)(Random.nextInt))
-  private val LoopPebbleBody = PebbleStringBody(resourceAsString("sample-peeble2.json", charset))
+  private val LoopPebbleBody = PebbleStringBody(resourceAsString("sample-peeble2.json", charset), charset)
```
Comment (slandelle) [inline gatling-benchmarks/src/main/scala/io/gatling/core/session/pebble/PebbleBenchmark.scala:37]: Oops! No idea why this didn't show up on CI.

File: gatling-http/src/main/scala/io/gatling/http/action/ws/WsAction.scala [hunk 2]
```diff
@@ -17,14 +17,18 @@
...
 
-  final def fetchFsm(wsName: String, session: Session): Validation[WsFsm] =
-    session.attributes.get(wsName) match {
-      case Some(wsFsm) => wsFsm.asInstanceOf[WsFsm].success
-      case _           => s"Couldn't fetch open webSocket".failure
-    }
+  final def fetchFsm(wsName: Expression[String], session: Session): Validation[WsFsm] =
+    wsName
+      .apply(session)
+      .flatMap(name => {
```
Comment (slandelle) [inline gatling-http/src/main/scala/io/gatling/http/action/ws/WsAction.scala:28]: Please remove extra parens:

```scala
.flatMap { name =>

}
```

File: gatling-http/src/main/scala/io/gatling/http/action/ws/WsConnect.scala [hunk 3]
```diff
@@ -20,6 +20,7 @@ import io.gatling.commons.util.Clock
 import io.gatling.commons.validation._
 import io.gatling.core.CoreComponents
 import io.gatling.core.action.{ Action, RequestAction }
+import io.gatling.core.session._
```
Comment (slandelle) [inline gatling-http/src/main/scala/io/gatling/http/action/ws/WsConnect.scala:23]: Weird. You're importing `session._` while the line below imports `session.{ Expression, Session }`. If you need something else, please remove the second line that becomes useless.

**Review by slandelle — CHANGES_REQUESTED** [review]
Thanks for contributing!

Please address my comments.

FYI, we're currently updating our CLA process, so I'll ping you when we're ready so we can proceed with it and then merge.

**Comment by jklingelAvid** [comment]
Thanks a lot for the review. I changed it according to your suggestion.

**Comment by slandelle** [comment]
Thanks! Could you please sign our [CLA](https://docs.google.com/forms/d/1hbpxVqJ5hIYYJuOjOyOzub9fJFGFGvvfnB5w5LCWdUw)?

Also, could you please fix the commit history and squash into a single one on top of master? If you really can't do that, I'll take care of it.

Cheers

**Comment by jklingelAvid** [comment]
@slandelle I messed up with the squash & rebase. I'm now creating another pull request with a clean branch. Sorry for the inconvenience

**Pull request closed without merging**